GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -482,6 +482,12 @@ public Result Status(bool allowObjectDownloads, bool useStatusCache, bool showUn
             return this.InvokeGitInWorkingDirectoryRoot(command, useReadObjectHook: allowObjectDownloads);
         }
 
+        public Result StatusPorcelain()

```

Comment:
Should we have some kind of helper method and basic struct so callers don't need to roll their own parser?

Reply:
I thought about that and ended up not doing that since this is the only case that needs this functionality right now.  If there ends up being another use case then I would definitely do something like that.  If you feel like this would be used in other places, then I will make it more generic.  I know there are places that we are running status but only checking for `nothing to commit, working tree clean` which is really the wrong thing to do and we should be using the porcelain command.  Let me know what you think.

Reply:
Let's leave that as cleanup for later (i.e. when status changes output and we are forced to switch to porcelain)